client/components/pack/PackDetails.js

```diff
@@ -21,42 +21,46 @@ import { AddItem } from "../item/AddItem";
 import { AddItemModal } from "./AddItemModal";
 
 export function PackDetails() {
-  const searchParams = new URLSearchParams(window.location.search);
-  let canCopy = searchParams.get("copy");
+  // const searchParams = new URLSearchParams(window.location.search);
+
+  // let canCopy = searchParams.get("copy");
 
   const dispatch = useDispatch();
 
   const { packId } = useSearchParams();
-
   const link = `${CLIENT_URL}/packs/${packId}`;
-
+  const isLoading = useSelector((state) => state.singlePack.isLoading);
+  const updated = useSelector((state) => state.packs.update);
+  const [firstLoad, setFirstLoad] = useState(true);
   const user = useSelector((state) => state.auth.user);
   const userId = user && user._id;
+  const [isAddItemModalOpen, setIsAddItemModalOpen] = useState(false);
+  const currentPack = useSelector((state) => state.singlePack.singlePack);
   useEffect(() => {
     if (!packId) return;
-    dispatch(fetchSinglePack(packId));
-    if (userId) dispatch(fetchUserPacks(userId));
-  }, [dispatch, packId]);
+      dispatch(fetchSinglePack(packId));
+      if (userId) dispatch(fetchUserPacks(userId));
+      setFirstLoad(false)
+  }, [dispatch, packId, updated]);
```

Comment:
Having updated here is kind of hacky, because now we refetch everytime we change item. we can leave this, but our store is also not properly updating with the data it gets back from backend CRUD call. That is our issue.